src/main/actions.js

```diff
@@ -50,8 +51,23 @@ export function setupActions(store) {
     openFileForImporting(undefined, type, archiveId);
   });
 
-  ipc.on('locale-change', () => {
-    i18n.setup(store);
+  ipc.on('change-locale-main', (e, lang) => {
+    let locale = lang;
+    if (!locale) {
+      const win = getMainWindow();
+      // set system lang
+      locale = app.getLocale().split('-')[0] || 'en';
+
+      // check if its available
+      if (Object.keys(languages).indexOf(locale) >= 0) {
```

Comment:
This block is wrong. It should be checking if its "not" available, then use default:

``` js
if (!(locale in languages)) {

Reply:
Even better solution (: thanks